docs: build and push the API docs to github pages

elopio opened the pull request:
Requires #1125

sergiusens commented on docs/api/Makefile:
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
why isn't this in `setup.py` instead?

elopio replied:
This is autogenerated by sphinx. I'd prefer not to write anything by myself, and just use the tool.

kyrofa commented:
CLA check here.

elopio commented:
@kyrofa I think I'll have to copy the changes into a new branch, because I don't know where it took my personal email and I don't know how to force it to take my work email. I deleted the branch, rebased to make a new single commit, and made the pr again, and it still fails.

kyrofa commented:
@elopio you probably made a `git config --global` with your personal details. Use `git config` within the snapcraft repo with your work details, then `git commit --amend --reset-author`.

elopio commented:
Thanks @kyrofa, --reset-author was the magical step.

sergiusens commented on tools/push_api_gh_pages.sh:
```diff
@@ -0,0 +1,20 @@
+
+set -ev
+
+GH_REPO="@github.com/elopio/snapcraft.git"
```
Can you push to a `gh-pages` branch on github.com:snapcore/snapcraft.git instead?

elopio replied:
@sergiusens done. But then you need to update the token in the travis settings, with permissions to repo and user.

sergiusens commented on tools/push_api_gh_pages.sh:
```diff
@@ -0,0 +1,20 @@
+
+FULL_REPO="https://$GH_TOKEN$GH_REPO"
+
+cd docs/api/_build/html
```
the plan was to delete `docs` and add `snappy-docs` as a submodule, can this be somewhere else?

elopio replied:
I moved it to tools.

sergiusens commented:
oh, and there are conflicts :-)
